src/mmap.cpp

```diff
@@ -117,6 +177,29 @@ file_handle::file_handle(string_view name, std::int64_t
 		DWORD temp;
 		::DeviceIoControl(m_fd, FSCTL_SET_SPARSE, nullptr, 0, nullptr, 0, &temp, nullptr);
 	}
+
+	if (mode & open_mode::truncate)
+	{
+		LARGE_INTEGER sz;
+		sz.QuadPart = size;
+		if (SetFilePointerEx(m_fd, sz, nullptr, FILE_BEGIN) == FALSE)
+			throw_ex<system_error>(error_code(GetLastError(), system_category()));
+
+		if (::SetEndOfFile(m_fd) == FALSE)
```

Comment:
calling `SetEndOfFile()` is redundant, as creating the file mapping will implicitly resize the file as well. This came along with the `SetFileValidData()` patch below. I think in order for that to work, the file needs to be extended first.